internal/provider/providers/nowdns/provider.go

```diff
@@ -0,0 +1,167 @@
+package nowdns
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/netip"
+	"net/url"
+	"strings"
+
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/provider/constants"
+	"github.com/qdm12/ddns-updater/internal/provider/errors"
+	"github.com/qdm12/ddns-updater/internal/provider/headers"
+	"github.com/qdm12/ddns-updater/internal/provider/utils"
+	"github.com/qdm12/ddns-updater/pkg/publicip/ipversion"
+)
+
+type Provider struct {
+	domain        string
+	host          string
+	ipVersion     ipversion.IPVersion
+	username      string
+	password      string
+	useProviderIP bool
+}
+
+func New(data json.RawMessage, domain, host string,
+	ipVersion ipversion.IPVersion) (p *Provider, err error) {
+	extraSettings := struct {
+		Username      string `json:"username"`
+		Password      string `json:"password"`
+		UseProviderIP bool   `json:"provider_ip"`
+	}{}
+	err = json.Unmarshal(data, &extraSettings)
+	if err != nil {
+		return nil, err
+	}
+	p = &Provider{
+		domain:        domain,
+		host:          host,
+		ipVersion:     ipVersion,
+		username:      extraSettings.Username,
+		password:      extraSettings.Password,
+		useProviderIP: extraSettings.UseProviderIP,
+	}
+	err = p.isValid()
+	if err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+func (p *Provider) isValid() error {
+	switch {
+	case p.username == "":
+		return fmt.Errorf("%w", errors.ErrUsernameNotSet)
+	case p.password == "":
+		return fmt.Errorf("%w", errors.ErrPasswordNotSet)
+	}
+	return nil
+}
+
+func (p *Provider) String() string {
+	return utils.ToString(p.domain, p.host, constants.NowDNS, p.ipVersion)
+}
+
+func (p *Provider) Domain() string {
+	return p.domain
+}
+
+func (p *Provider) Host() string {
+	return p.host
+}
+
+func (p *Provider) IPVersion() ipversion.IPVersion {
+	return p.ipVersion
+}
+
+func (p *Provider) Proxied() bool {
+	return false
+}
+
+func (p *Provider) BuildDomainName() string {
+	return utils.BuildDomainName(p.host, p.domain)
+}
+
+func (p *Provider) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain:    fmt.Sprintf("<a href=\"http://%s\">%s</a>", p.BuildDomainName(), p.BuildDomainName()),
+		Host:      p.Host(),
+		Provider:  "<a href=\"https://www.now-dns.com/\">Now-DNS</a>",
+		IPVersion: p.ipVersion.String(),
+	}
+}
+
+func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Addr) (newIP netip.Addr, err error) {
+	u := url.URL{
+		Scheme: "https",
+		Host:   "now-dns.com",
+		Path:   "/update",
+		User:   url.UserPassword(p.username, p.password),
+	}
+
+	values := url.Values{}
+	values.Set("hostname", utils.BuildURLQueryHostname(p.host, p.domain))
```

Comment:
Just have this as `p.domain`, so it won't be as `.domain.com` if the host is empty.

```suggestion
	values.Set("hostname", p.domain)
```

Reply:
Vicariously fixed this. See below conversation and commit https://github.com/qdm12/ddns-updater/pull/504/commits/e6bb0f37dd4b76b33036e73edce4297a99083afd